pkg/service/catalog.go

```diff
@@ -176,8 +180,19 @@ func NewService(
 		plans:             plans,
 		indexedPlans:      make(map[string]Plan),
 	}
-	for _, plan := range s.plans {
-		s.indexedPlans[plan.GetID()] = plan
+	for _, planIfc := range s.plans {
+		p := planIfc.(*plan)
+		if serviceProperties.ParentServiceID == "" {
+			if p.ParameterSchemas == nil {
+				p.ParameterSchemas = &planSchemas{}
+			}
+			p.ParameterSchemas.addParameters(
```

Comment:
In schema.go, I've created two "addParameters()" methods. This one is on a *planSchemas receiver and takes in the three types of param schemas we can have now. I wanted to contain/centralize all that logic so it was reuable in different spots and we can mostly replace the unexported structs with maps later and just update those methods.